src/libraries/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/SymmetricAlgorithm.cs

```diff
@@ -428,6 +428,317 @@ public int GetCiphertextLengthCfb(int plaintextLength, PaddingMode paddingMode =
+        /// <param name="bytesWritten">When this method returns, the total number of bytes written to <paramref name="destination" /></param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="paddingMode" /> is not a valid padding mode.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <see cref="TryEncryptEcbCore" /> could not encrypt the plaintext.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   The buffer in <paramref name="destination"/> is too small to hold the ciphertext data.
+        /// </exception>
+        /// <remarks>
+        ///   This method's behavior is defined by <see cref="TryEncryptEcbCore" />.
+        /// </remarks>
+        public bool TryEncryptEcb(ReadOnlySpan<byte> plaintext, Span<byte> destination, PaddingMode paddingMode, out int bytesWritten)
+        {
+            CheckPaddingMode(paddingMode);
+            return TryEncryptEcbCore(plaintext, destination, paddingMode, out bytesWritten);
+        }
+
+        /// <summary>
+        ///   When overridden in a derived class, attempts to encrypt data using ECB
+        ///   mode with the specified padding mode in to a buffer.
+        /// </summary>
+        /// <param name="plaintext">The data to encrypt</param>
+        /// <param name="destination">The buffer to receive the ciphertext data</param>
+        /// <param name="paddingMode">The padding mode used to produce the ciphertext and remove during decryption</param>
+        /// <param name="bytesWritten">When this method returns, the total number of bytes written to <paramref name="destination" /></param>
+        /// <exception cref="NotSupportedException">
+        ///   A derived class has not provided an implementation.
+        /// </exception>
+        /// <remarks>
+        ///   Derived classes must override this and provide an implementation.
+        /// </remarks>
+        protected virtual bool TryEncryptEcbCore(
+            ReadOnlySpan<byte> plaintext,
+            Span<byte> destination,
+            PaddingMode paddingMode,
+            out int bytesWritten)
+        {
+            throw new NotSupportedException(SR.NotSupported_SubclassOverride);
+        }
+
+        /// <summary>
+        ///   When overridden in a derived class, attempts to decrypt data using ECB
+        ///   mode with the specified padding mode in to a buffer.
+        /// </summary>
+        /// <param name="ciphertext">The data to decrypt</param>
+        /// <param name="destination">The buffer to receive the plaintext data</param>
+        /// <param name="paddingMode">The padding mode used to produce the ciphertext and remove during decryption</param>
+        /// <param name="bytesWritten">When this method returns, the total number of bytes written to <paramref name="destination" /></param>
```

Comment:
(Added trailing periods and missing `<returns>`)

```suggestion
        /// <param name="ciphertext">The data to decrypt.</param>
        /// <param name="destination">The buffer to receive the plaintext data.</param>
        /// <param name="paddingMode">The padding mode used to produce the ciphertext and remove during decryption.</param>
        /// <param name="bytesWritten">When this method returns, the total number of bytes written to <paramref name="destination" />.</param>
        /// <returns><see langword="true"/> if <paramref name="destination"/> was large enough to receive the decrypted data; otherwise, <see langword="false" />.</returns>
```